### ISSUE TITLE
error: cannot produce cdylib for `fastuuid v0.6.0 (/tmp)` as the target `x86_64-unknown-linux-musl` does not support these crate types

### ISSUE THREAD

**Issue by thedrow** [PyO3/maturin#427]
Please provide the following information: 
 * Your python version (`python -V`): Python 3.8.7
 * Your pip version (`pip -V`): pip 20.3.4 from /usr/lib/python3.8/site-packages/pip (python 3.8)
   * If this isn't the [latest pip version](https://pip.pypa.io/en/stable/news/), have you checked using the latest pip version?
 * The version of the bindings you're using, if any (e.g. `pyo3`, `rust-cpython` or `cffi`): pyo3
 * Does `cargo build` work? No
 * If on windows, have you checked that you aren't accidentally using unix path (those with the forward slash `/`)?

Please list the exact steps required to reproduce your error with all command output and if possible with a repository:
Compile any pyo3 library on Alpine Linux.

**Comment by konstin** [comment]
Sorry, I accidentally disimproved that

**Comment by thedrow** [comment]
> 
> 
> Sorry, I accidentally disimproved that

It must be the lack of caffaine but what are you referring to by "disimproved"?

**Comment by thedrow** [comment]
I still have the same problem.
```
bash-5.1# maturin build -i python3
🔗 Found pyo3 bindings
🐍 Found CPython 3.8 at python3
📦 Built source distribution to /tmp/target/wheels/fastuuid-0.6.0.tar.gz
error: cannot produce cdylib for `fastuuid v0.6.0 (/tmp)` as the target `x86_64-unknown-linux-musl` does not support these crate types
💥 maturin failed
  Caused by: Failed to build a native library through cargo
  Caused by: Cargo build finished with "exit code: 101": `cargo rustc --message-format json --manifest-path Cargo.toml --lib --`
bash-5.1# maturin --version
maturin 0.9.4
```
please reopen.

**Comment by konstin** [comment]
> It must be the lack of caffaine but what are you referring to by "disimproved"?

messense wrote the correct patch using `RUSTFLAGS`, I thought I'll be smarter by using the `cargo rustc` command line arguments instead, but since I failed to test it broke musl again. I've added yet another fix in 7103f037a3d0d89a621e5358b0c9882b634a6cd4 and a test to catch regressions.

**Comment by thedrow** [comment]
I can confirm that this works.
Is there a chance for a new bugfix release?

**Comment by konstin** [comment]
I've published 0.10.0-beta.1 which you could pin until the stable release
